src/workerd/api/global-scope.c++

```diff
@@ -247,10 +249,12 @@ kj::Promise<DeferredProxy<void>> ServiceWorkerGlobalScope::request(
     return ioContext.awaitJs(lock ,promise.then(kj::implicitCast<jsg::Lock&>(lock),
         ioContext.addFunctor(
             [&response, allowWebSocket = headers.isWebSocket(),
-             canceled = kj::addRef(*canceled), &headers]
-            (jsg::Lock& js, jsg::Ref<Response> innerResponse)
+             canceled = kj::addRef(*canceled), &headers, span = kj::mv(span)]
+            (jsg::Lock& js, jsg::Ref<Response> innerResponse) mutable
             -> IoOwn<kj::Promise<DeferredProxy<void>>> {
       auto& context = IoContext::current();
+      // Drop our fetch_handler span now that the promise has resolved.
+      span = kj::none;
```

Comment:
You can also call `span.end()`, which means you don't have to wrap it in a maybe. That way in the future if you ever want to add logs or tags you won't have to `IF_SOME` it out.

Reply:
I've added a comment to the ticket internally to apply this suggestion if we get around to adding logs in the future. I think we just want to get the change out for now.

Reply:
Sure, either approach is fine.  Another option is to just let the span end implicitly when the lambda closure is destroyed (as will also happen when the promise is cancelled or rejected).  The main benefit of explicitly ending the span early is that you'd omit the cost of the rest of the lambda execution -- which should be marginal in the common case, but being explicit might help debug the rare situations where it isn't?